src/transports/ws.rs

```diff
@@ -139,6 +139,20 @@ impl WsServerTask {
             resource
         );
         let mut client = Client::new(socket, host, &resource);
+        let maybe_encoded = url.password().map(|password| {
+            use headers::authorization::{Authorization, Credentials};
+            log::trace!("Basic auth username: {}", url.username());
+            log::trace!("Basic auth password: {}", password);
+            Authorization::basic(url.username(), password).0.encode().to_str().unwrap().as_bytes().to_owned()
```

Comment:
```suggestion
            Authorization::basic(url.username(), password).0.encode().as_bytes().to_vec()
```
I don't think it's necessary to go via `&str` first, especially given we just unwrap.